### ISSUE TITLE
`:normal` command does not act on original range of lines if the command itself adds lines

### ISSUE THREAD

**Issue by famiu** [neovim/neovim#24961]
### Problem

If I'm working on a file like this:
```
{foo,bar,baz},
{bar,baz,foo},
{baz,foo,bar},
{foo,baz,bar},
{bar,foo,baz},
{baz,bar,foo},
```

and I want to split every list so that every element is on its own line, so I execute this command:
```
:execute "1,6normal ^a\<CR>\<Tab>\<Esc>f,a\<CR>\<Esc>f,a\<CR>\<Esc>f}i\<CR>\<BS>"
```

I'd expect a result that is the same as what I would get if I were to execute the command on each line separately, something like:
```
{
	foo,
	bar,
	baz
},
{
	bar,
	baz,
	foo
},
{
	baz,
	foo,
	bar
},
{
	foo,
	baz,
	bar
},
{
	bar,
	foo,
	baz
},
{
	baz,
	bar,
	foo
},
```

but instead, I get a result like:
```
{
	f
		o
			o
				,
					
						



	bar,
	baz
},
{bar,baz,foo},
{baz,foo,bar},
{foo,baz,bar},
{bar,foo,baz},
{baz,bar,foo},
```

Which makes it clear that `:normal` executes the command on the added lines, instead of the original set of lines that were the target of the command. This to me does not seem like sensible behavior.

### Steps to reproduce

1. `nvim --clean`
2. Open the example text in the above section
3. Execute the command given in the above section
4. See unexpected result

### Expected behavior

`:normal` should instead work on the initially targeted range of lines instead of acting on newly added lines, somewhat similar to what `g/pattern/cmd` does.

### Neovim version (nvim -v)

NVIM v0.10.0-dev-837+g389165cac

### Vim (not Nvim) behaves the same?

yes, vim 9.0

### Operating system/version

Fedora 38

### Terminal name/version

GNOME Terminal 3.48.1

### $TERM environment variable

xterm-256color

### Installation

Build from repo

**Comment by zeertzjq** [comment]
The expected behavior can be easily achieve by adding `:global/.*/` before the `:normal`.
```vim
:execute "1,6global/.*/normal ^a\<CR>\<Tab>\<Esc>f,a\<CR>\<Esc>f,a\<CR>\<Esc>f}i\<CR>\<BS>"
```
So I don't think there is a good enough reason to change the behavior of `:normal` with a range.

**Comment by lewis6991** [comment]
Interesting. Out of interest could you ELI5 how/why this works?

**Comment by zeertzjq** [comment]
It's in the docs below `:h :global`.
